File: IsomorphicString.java
// Time Complexity : 
// Space Complexity :
// Did this code successfully run on Leetcode :
// Three line explanation of solution in plain english

// Your code here along with comments explaining your approach


class Solution {
    public boolean isIsomorphic(String s, String t) {
       if(s.length() != t.length())    return false;
       HashMap<Character, Character> map = new HashMap<>();
       HashSet<Character> set = new HashSet<>();
       for(int i=0; i<s.length(); i++)
       {
           char c = s.charAt(i);
           char d = t.charAt(i);
           if(!map.containsKey(c))
           {
               if (set.contains(d))
                   return false;
               map.put(c, d);
               set.add(d);
           }
           else
           {
               if(map.get(c) != d)
                   {set.add(d);
                   return false;}
           }
       }
       return true;
}
}

File: IsomorphicString.cs
//TC:O(N)
//SC:O(N)
public class Solution {
    public bool IsIsomorphic(string s, string t) {
        Dictionary<char,char> st = new Dictionary<char,char>();
        Dictionary<char,char> ts = new Dictionary<char,char>();
        
        for(int i=0;i<s.Length;i++){
            char sCh = s[i];
            char tCh = t[i];
            
            if(st.ContainsKey(sCh)){
                if(st[sCh]!=tCh){
                    return false;
                }
            }else{
                st.Add(sCh,tCh);
            }
            
             if(ts.ContainsKey(tCh)){
                if(ts[tCh]!=sCh){
                    return false;
                }
            }else{
                ts.Add(tCh,sCh);
            }
        }
        return true;
    }
}